packages/node_modules/@ciscospark/spark-core/src/lib/batcher.js

```diff
@@ -167,11 +168,16 @@ const Batcher = SparkPlugin.extend({
    * @returns {Promise<undefined>}
    */
   handleHttpError(reason) {
-    const msg = reason.message || reason.body || reason;
-    return Promise.all(reason.options.body.map((item) => this.getDeferredForRequest(item)
-      .then((defer) => {
-        defer.reject(msg);
-      })));
+    if (reason instanceof SparkHttpError) {
+      if (has(reason, `options.body.map`)) {
+        return Promise.all(reason.options.body.map((item) => this.getDeferredForRequest(item)
+          .then((defer) => {
+            defer.reject(reason);
+          })));
+      }
+    }
+    this.logger.error(`http error handler called without a SparkHttpError object`);
+    return Promise.reject(new Error(`An undefined http error has occurred.`));
```

Comment:
ok, saw one more issue; if reason is not a `SparkHttpError`, it may still carry useful information. we should certainly log it and can probably just reject with it directly.